eval/eval_adapter.py

```diff
@@ -0,0 +1,362 @@
+
+        # Returns a list containing all values of the task registry that
+        # match at least one of the patterns
+        import fnmatch
+
+        def pattern_match(patterns, source_list):
+            task_names = set()
+            for pattern in patterns:
+                for matching in fnmatch.filter(source_list, pattern):
+                    task_names.add(matching)
+            return list(task_names)
+
+        eval_tasks = pattern_match(eval_tasks, tasks.ALL_TASKS)
+        print(f"Found tasks: {eval_tasks}")
+
+        # **HACK INCOMING**:
+        # first get task dict on local main rank
+        # the tasks are downloaded *as they are initialized*, and the downloads don't like multithreading.
+        # so we download them once on the local main rank, wait, and then initialize them on all other ranks, which *should* load from the cache.
+        if self.fabric.local_rank == 0:
+            tasks.get_task_dict(eval_tasks)
+        # torch barrier
+        self.fabric.barrier()
+        task_dict = tasks.get_task_dict(eval_tasks)
+
+        lm = self
+        if use_cache:
+            lm = base.CachingLM(lm, "lm_cache/" + name + ".db")
+
+        results = evaluator.evaluate(
+            lm=lm,
+            task_dict=tasks.get_task_dict(eval_tasks),
+            description_dict=description_dict,
+            num_fewshot=num_fewshot,
+            limit=limit,
+            bootstrap_iters=bootstrap_iters,
+        )
+
+        results["config"] = {
+            "model": self.model.config.name,
+            "num_fewshot": num_fewshot,
+            "batch_size": self.batch_size,
+            "device": str(self.device),
+            "no_cache": not use_cache,
+            "limit": limit,
+            "bootstrap_iters": bootstrap_iters,
+            "description_dict": description_dict,
+        }
+
+        return results
+
+
+def run_eval_harness(
+    checkpoint_dir: str = "",
+    precision: str = "bf16-true",
+    batch_size=1,
+    eval_tasks: Optional[List[str]] = None,
+    num_fewshot=0,
+    bootstrap_iters=2,
+    temperature=1.0,
+    device="auto",
+    devices: int = 1,
+    strategy: str = "auto",
+    quantize: Optional[
+        Literal[
+            "bnb.nf4",
+            "bnb.nf4-dq",
+            "bnb.fp4",
+            "bnb.fp4-dq",
+            "bnb.int8",
+            "gptq.int4",
+        ]
+    ] = None,
+):
+    adapter = EvalHarnessAdapter(
+        checkpoint_dir=checkpoint_dir,
+        precision=precision,
+        batch_size=batch_size,
+        temperature=temperature,
+        device=device,
+        devices=devices,
+        strategy=strategy,
+        quantize=quantize,
+    )
+    adapter.fabric.print("Running evaluation harness...")
+    results = adapter.run_eval(
+        eval_tasks=eval_tasks,
+        num_fewshot=num_fewshot,
+        bootstrap_iters=bootstrap_iters,
+        use_cache=False,
+    )
+    data = json.dumps(results)
+    filename = str(time.time()) + "-eval.txt"
```

Comment:
Is it a convention to use that date/time format? Maybe some OS might sort weirdly if the timing is not in this format (how it's called, ticks perhaps)? Or maybe then the date/time is parsed from the name of the file?
I know it's a small, really-really small issue, but I prefer to see more obvious date/time format in the filename:

<img width="716" alt="Screenshot 2023-08-06 at 7 06 15 PM" src="https://github.com/Lightning-AI/lit-gpt/assets/58434077/1b802a61-b5f8-4797-ab7a-bc4d026a646b">

Reply:
`time.time()` value is same everywhere in the world when called at the same moment. my intent here was only to have a unique filename (sortable with timestamp).